go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/Shopify/sarama
 
 go 1.16
 
+retract [v1.31.0, v1.31.2] // https://github.com/Shopify/sarama/issues/2129
```

Comment:
It's a range so it includes "1.31.1" as well. The idea is that if you merge this and release it as "1.31.2" then when a user uses go tooling to "upgrade" or get the latest version, because 1.31.0, 1.31.1 and 1.31.2 are retracted, 1.30.1 will be considered as the latest. 

So for example if a user is currently using 1.31.1 they will be "upgraded" to 1.30.1. 

After you release the fix in 1.31.3 everything will be back to normal.